### PR TITLE
DDF-5439 Decouple UI and Service Facet Whitelist

### PR DESCRIPTION
**Pull request by willwill96**
#### What does this PR do?
Breaks Facet Whitelist into two whitelists; one at the service level and one for the UI
#### Who is reviewing it? 
@andrewzimmer 
@hayleynorton 
@zta6 

#### Ask 2 committers to review/merge the PR and tag them here.
@djblue 
@bdeining 

#### How should this be tested?
1. Build / Install / Ingest a document or two
2. In the admin console, add some text attribute names to `Catalog UI Search Attribute Suggestion List`
3. Refresh Intrigue. Select some of those attributes from query advanced. Verify that the input is NOT populated with a dropdown on those attributes
4. In the admin console, add some text attribute names to `Facet Attribute Whitelist` (with some overlap from step 2)
5. Refresh Intrigue. Select some of those attributes from query advanced. Verify the ones that were added to both configs populate the input with a dropdown. Verify the others do not.
6. In postman query the `/internal/cql` endpoint and verify that all attributes from step 4 are facetable. An example request looks like:
```
{
    "cql": "anyText ILIKE '%'",
    "facets": ["title"]
}
``` 
7. Any other spot testing appropriate.

#### What are the relevant tickets?
Fixes: #5439 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/ui/catalog-ui-search/src/main/resources/OSGI-INF/metatype/org.codice.ddf.catalog.ui.facetwhitelist.xml [hunk 1]
```diff
@@ -15,11 +15,11 @@
...
 
-    <OCD name="Catalog UI Search Attribute Suggestion Whitelist" id="org.codice.ddf.catalog.ui.facetwhitelist">
+    <OCD name="Facet Attribute Whitelist" id="org.codice.ddf.catalog.ui.facetwhitelist">
         <AD id="facetAttributeWhitelist"
-            name="Attribute Suggestion Whitelist"
-            description="List of attributes that should provide autocomplete suggestions. Suggestions are typically popular values for the attribute in data storage.
-Caution: Suggestion values are not protected by any security. Only choose attributes whose values will be safe for all users to view."
+            name="Facet Attribute Whitelist"
+            description="Attributes that can be faceted against through the catalog framework.
+Caution:Results from faceted queries are not protected by any security. Only choose attributes whose values will be safe for all users to view."
```
Comment (hayleynorton) [inline catalog/ui/catalog-ui-search/src/main/resources/OSGI-INF/metatype/org.codice.ddf.catalog.ui.facetwhitelist.xml]: ✏️ do you want a space here after the colon?

File: ui/packages/catalog-ui-search/src/main/webapp/component/query-advanced/query-advanced.view.js [hunk 2]
```diff
@@ -50,7 +50,7 @@ const isValidFacetAttribute = (id, type) => {
   if (id === 'anyText') {
     return false
   }
-  if (!properties.facetWhitelist.includes(id)) {
+  if (!properties.attributeSuggestionList.includes(id)) {
```
Comment (bdeining) [inline ui/packages/catalog-ui-search/src/main/webapp/component/query-advanced/query-advanced.view.js]: just to confirm, this is the only area on the front end that uses the `facetWhitelist` property?

Comment (willwill96) [reply ui/packages/catalog-ui-search/src/main/webapp/component/query-advanced/query-advanced.view.js]: Yep

**Comment by hayleynorton** [comment]
Hero successful
Verified that attributes on the suggestion list but not on the whitelist **did not** populate with suggestions, attributes on the whitelist but not on the suggestion list **did not** populate with suggestions, and attributes on both lists **did** populate with suggestions. Tested with attributes that were present on at least one result. Attributes on both lists but not present on any results did not populate with suggestions as expected.

**Comment by andrewzimmer** [comment]
build now

**Comment by willwill96** [comment]
build now

**Comment by willwill96** [comment]
build now

**Comment by willwill96** [comment]
build now

**Comment by andrewzimmer** [comment]
Hero Successful 🎉 
same changes verified in 2.18 and 2.19

**Pull request merged**